ansible_specdoc/objects.py

```diff
@@ -134,10 +213,36 @@ def doc_dict(self) -> Dict[str, Any]:
         result['options'] = {k: v.doc_dict for k, v in self.options.items() if not v.doc_hide}
 
         if self.return_values is not None:
-            result['return_values'] = {k: v.__dict__ for k, v in self.return_values.items()}
+            result['return_values'] = {k: v.doc_dict for k, v in self.return_values.items()}
```

Comment:
When I pass a dict to return_values like this:
```python
SPECDOC_META = SpecDocMeta(
    [...]
    # shuold be return_values=SpecReturnValue(
    return_values=dict(
        id=dict(
            description='UUID of the project.',
            docs_url='https://metal.equinix.com/developers/api/projects/#retrieve-a-project',
            type='str',
            sample='eef49903-7a09-4ca1-af67-4087c29ab5b6',
        ),
```
I get exception:

```
    result['options'] = {k: v.doc_dict for k, v in self.options.items() if not v.doc_hide}
AttributeError: 'dict' object has no attribute 'doc_hide'
```

Would it be possible to use typing to explicitly say that return_values expect SpecReturnValue? Or do I need to call mypy or something to do the type checks?

Reply:
You'll probably need to use mypy for the type checks in this case. It previously accepted `Dict[str, Dict]` but we've since started requiring `Dict[str, SpecReturnValue]` so its easier to validate your spec before runtime :+1:

Reply:
thanks!